src/ser.rs

```diff
@@ -126,17 +126,26 @@ impl ser::SerializeTupleStruct for ArraySerializer<'_> {
     }
 }
 
+pub enum MapResult {
+    Map(Map),
+    Object(Object),
```

Comment:
I guess that custom `Object` wrapper has to be public?

```
warning: private type `ser::Object` in public interface (error E0446)
   --> src/ser.rs:131:12
    |
131 |     Object(Object),
    |            ^^^^^^
    |
    = note: `#[warn(private_in_public)]` on by default
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>
```

Reply:
You can use `js_sys::Object` and convert one into another with `.unchecked_into()` (we know they're just bindings to the same type so it's safe to do so).